mlflow/pytorch/__init__.py

```diff
@@ -571,3 +571,16 @@ def predict(self, data, device="cpu"):
             predicted = pd.DataFrame(preds.numpy())
             predicted.index = data.index
             return predicted
+
+
+def autolog(log_every_n_iter=1):
+    """
+    Wrapper for `mlflow.pytorch._pytorch_autolog.autolog` method.
+    Only PyTorch lightning is supported for this autolog feature.
```

Comment:
@smurching Mentioned about the support here. Please check if this docstring looks fine or we need more detailed description on this.

Reply:
Thanks @shrinath-suresh! For this PR, how about:

```suggestion
    Automatically log metrics, params, and models from `PyTorch Lightning <https://pytorch-lightning.readthedocs.io/en/latest>`_ model training.
    **Note**: Autologging is only supported for PyTorch Lightning models, i.e. models that subclass `pytorch_lightning.LightningModule <https://pytorch-lightning.readthedocs.io/en/latest/lightning_module.html>`_. In particular, autologging support for vanilla Pytorch models that only subclass `torch.nn.Module <https://pytorch.org/docs/stable/generated/torch.nn.Module.html>`_ is not yet available.
```

Either in this PR or in a follow-up, we should enhance this docstring to describe when autologging is invoked & what exactly gets autologged. See the [sklearn autologging docstring](https://github.com/mlflow/mlflow/blob/37496c3161a9f7561b4c7ca3054ae6bdb3154523/mlflow/sklearn/__init__.py#L537-L703) for an example. We should also make sure to sure to mention Pytorch autologging in the "automatic logging" section within the MLflow Tracking API docs: https://github.com/mlflow/mlflow/blob/master/docs/source/tracking.rst#id70

Reply:
@smurching This is being handled in the Doc PR separately